include/zephyr/drivers/espi_emul.h

```diff
@@ -116,20 +53,65 @@ struct espi_emul {
 	/** Parent emulator */
 	const struct emul *parent;
 	/** API provided for this device */
-	const struct emul_espi_device_api *api;
+	const struct espi_emul_device_api *api;
 	/** eSPI chip-select of the emulated device */
 	uint16_t chipsel;
 };
 
-/** Definition of the eSPI controller emulator API */
-struct emul_espi_driver_api {
-	/* The struct espi_driver_api has to be first in
-	 * struct emul_espi_driver_api to make pointer casting working
+/** Data about the virtual wire */
+struct espi_emul_vw_data {
+	/* Virtual wire signal */
+	enum espi_vwire_signal sig;
+	/* The level(state) of the virtual wire */
+	uint8_t level;
+	/* The direction of the virtual wire. Possible values are
+	 * ESPI_MASTER_TO_SLAVE or ESPI_SLAVE_TO_MASTER
 	 */
-	struct espi_driver_api espi_api;
-	/* The rest, emulator specific functions */
-	emul_trigger_event trigger_event;
-	emul_find_emul find_emul;
+	uint8_t dir;
+};
+
+struct espi_emul_device_api {
```

Comment:
Can you comment the expected behavior, parameters, return value, etc. of each of these API functions? I see comments on at least some of the implementations. I'd prefer to have comments here and no comments there, since the interface is here.

Reply:
These functions are the same as for normal eSPI device, they are only required because the first parameter id espi_emul instead of device. Do we want to duplicate the comments?

Reply:
I guess not, but maybe include a reference to `struct espi_driver_api` here to point people like me in the right direction.

Reply:
I've updated the documentation, please take a look if it is better now

Reply:
The doc changes looks good.

As a nit, I think it would be helpful to group the members of this struct by who is expected to call them:

1. The EC. These would be the functions that have a direct analog in `espi_driver_api`, in the order they appear there.
2. The emulated bus. I think this would be `io_read` and `io_write`, but I'm not exactly clear on where those will be called from.
3. The emulated host. I think these are `internal_set_vwire` and `raise_event`.

Reply:
The io_* and set_vwire and raise_event are all for the host emulator. With the updated documentation, this order looks nice, done.